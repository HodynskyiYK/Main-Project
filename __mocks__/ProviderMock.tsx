import React from 'react'
import {MemoryRouter} from 'react-router-dom'
import { MovieModalContextProvider } from '../src/context/movie-modal-context'
import { Provider } from 'react-redux'
import { store } from '../src/store'
import { MovieDetailsContextProvider } from '../src/context/movie-details-context'

// @ts-ignore
export const ProvidersMock = ({ children }) => {
    return (
        <Provider store={store}>
            <MovieModalContextProvider>
                <MovieDetailsContextProvider>
                    <MemoryRouter>{children}</MemoryRouter>
                </MovieDetailsContextProvider>
            </MovieModalContextProvider>
        </Provider>
    )
}