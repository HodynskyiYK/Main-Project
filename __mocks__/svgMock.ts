// Mock Svg files
// https://github.com/gregberge/svgr/issues/83#issuecomment-437342861
const SvgMock = "icon-mock"
export const ReactComponent = SvgMock
export default SvgMock