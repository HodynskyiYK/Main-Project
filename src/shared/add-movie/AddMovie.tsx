import React, { FC } from 'react'
import { Button, ButtonTypes } from '../../components/button'
import { useMovieModalContext } from '../../context/movie-modal-context'
import { MovieModalForm } from '../movie-modal-form'

export const AddMovie: FC = () => {
    const {addMovieModalState, showAddMovieModal} = useMovieModalContext()


    return (
        <>
            <Button
                buttonClassName={ButtonTypes.ADD_BTN}
                buttonText={'+ add movie'}
                buttonAction={showAddMovieModal}
            />
            {
                addMovieModalState && <MovieModalForm modalName={'Add movie'}/>
            }
        </>
    )
}
