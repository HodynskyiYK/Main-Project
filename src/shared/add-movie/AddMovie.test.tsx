import React from 'react'
import { render } from '@testing-library/react'
import { AddMovie } from './AddMovie'
import { ProvidersMock } from '../../../__mocks__/ProviderMock'

describe('AddMovie tests', () => {

    test('To match snapshot', () => {
        const {asFragment} = render(<ProvidersMock><AddMovie/></ProvidersMock>)
        expect(asFragment()).toMatchSnapshot()
    })
})