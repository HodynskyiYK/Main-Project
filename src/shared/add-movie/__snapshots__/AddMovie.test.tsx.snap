// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMovie tests To match snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn addBtn"
  >
    + add movie
  </button>
</DocumentFragment>
`;
